index.js

```diff
@@ -774,12 +774,25 @@
   //. > S.ap([Math.sqrt, x => x * x], [1, 4, 9, 16, 25])
   //. [1, 2, 3, 4, 5, 1, 16, 81, 256, 625]
   //.
-  //. > S.ap(s => n => s.slice(0, n), s => Math.ceil(s.length / 2))('Haskell')
-  //. 'Hask'
-  //.
   //. > S.ap(S.Just(Math.sqrt), S.Just(64))
   //. Just(8)
   //. ```
+  //.
+  //. Note what happens when `Apply f => f` is replaced with `Function x`:
+  //.
+  //.     Apply f => f (a -> b) -> f a -> f b
+  //.     Function x (a -> b) -> Function x a -> Function x b
+  //.     Function x (a -> c) -> Function x a -> Function x c
+  //.     Function x (b -> c) -> Function x b -> Function x c
+  //.     Function a (b -> c) -> Function a b -> Function a c
+  //.     (a -> b -> c) -> (a -> b) -> (a -> c)
+  //.
+  //. This is the S combinator from combinatory logic.
```

Comment:
Good idea. "This" in the sentence above refers to `(a -> b -> c) -> (a -> b) -> (a -> c)` rather than to `ap`, which is more general. We could, though, combine this sentence with the sentence preceding the substitutions:

_Replacing `Apply f => f` with `Function x` produces the S combinator from combinatory logic:_